backend/Origam.Schema.LookupModel/SchemaItems/NewRecordScreenBinding.cs

```diff
@@ -48,7 +49,43 @@ public class NewRecordScreenBinding : AbstractSchemaItem, IAuthorizationContextC
     public NewRecordScreenBinding(Guid schemaExtensionId) : base(schemaExtensionId) { }
     
     public NewRecordScreenBinding(Key primaryKey) : base(primaryKey)	{}
-    
+
+
+    public bool IsAvailable
+    {
+        get
+        {
+            if (MenuItem is not FormReferenceMenuItem referenceMenuItem ||
+                referenceMenuItem.ReadOnlyAccess)
+            {
+                return false;
+            }
+
+            IParameterService parameterService = ServiceManager.Services
+                .GetService<IParameterService>();
+            IOrigamAuthorizationProvider authorizationProvider =
+                SecurityManager.GetAuthorizationProvider();
+            IPrincipal principal = SecurityManager.CurrentPrincipal;
+
+            string authContext = SecurityManager
+                .GetReadOnlyRoles(referenceMenuItem.AuthorizationContext);
+            bool hasReadOnlyRole = SecurityManager
+                .GetAuthorizationProvider()
+                .Authorize(SecurityManager.CurrentPrincipal, authContext);
+            if (hasReadOnlyRole)
+            {
+                return false;
+            }
+
+            return
```

Comment:
Why don't we use `FormTools.IsFormMenuReadOnly`?

Reply:
Because project `Origam.Gui` where the class `FormTools` lives cannot be referenced from `Origam.Workbench.Services` (where the `DataLookupService` lives) or from `Origam.Schema.LookupModel` (where the `NewRecordScreenBinding` lives). I did not think it was worth moving the `FormTools` to reuse two lines of code. Should I move it?

Reply:
No.